salt/metalk8s/internal/preflight/check.sls

```diff
@@ -0,0 +1,29 @@
+{%- from "metalk8s/map.jinja" import kubeadm_preflight with context %}
+
+{%- set status = {'error': False} %}
+{%- for sysctl, expected_value in kubeadm_preflight.mandatory.sysctl_values.items() %}
+  {%- set value = salt.sysctl.get(sysctl).strip('"') %}
+Check sysctl {{ sysctl }} value is {{ expected_value }}:
+  test.configurable_test_state:
+    - changes: False
+    - comment: {{ sysctl }} value is {{ value }}
+  {%- if value|int != expected_value|int %}
+  {%- do status.update({'error': True}) %}
+    - result: False
+  {%- endif %}
+{%- endfor %}
+
+All sysctl parameters are well configured:
+  test.configurable_test_state:
+    - changes: False
+    - comment: |-
+{%- if status.error %}
+        Some sysctl parameters have been set during installation, but
+        their values are not taken in account, please check the sysctl
+        configuration to ensure MetalK8s one have precedence.
+        See above to find parameters with wrong values.
+    - result: False
+    - failhard: True
+{%- else %}
+        Success!
+{%- endif %}
```

Comment:
Does it works ? For me, jinja is rendered at the same time as the one in `mandatory.sls` so you will not get up to date values
I think you need to use slot here (and btw slot will not be enough you need a custom module/state)
I would say a simple module to compare sysctl that just return True or false and then a slot on `result` in `test.configurable_test_state` (like we do in salt minion running state)

Also to me this state could be part of  `mandatory.sls` no ?

Reply:
Yes, you're right, I'm currently changing this.
I'm not sure it should be in `mandatory.sls`, it depends if we want it to be callable without doing any change which was the idea (so we can launch this on a regular basis through beacon/cron/whatever and send alert when it fails).

Reply:
Right, yes it can be useful but to me, it's not the goal of a salt state if we want something like that it should be a module directly instead I think.
Maybe you should create a simple module `metalk8s_checks.py` with a function for sysctl that basically do exactly the same as this state and then in mandatory state you just do a call to this module, what do you think ? (and then in the futur we could imagine adding other checks in this modules if needed)

Reply:
slowly but steadily building a platformcheck clone ;p. btw did you consider having the mandatory sysctl settings in /etc/sysctl.conf directly ? that way they re not overwritten (I guess)

Reply:
@TeddyAndrieux haha, that's what I'm currently doing. :)
@sayf-eddine-scality we could do that also, it works for sure.

Reply:
This is the order the sysctl are read:
```
/run/sysctl.d/*.conf
/etc/sysctl.d/*.conf
/usr/local/lib/sysctl.d/*.conf
/usr/lib/sysctl.d/*.conf
/lib/sysctl.d/*.conf
/etc/sysctl.conf
```
The last value defined overwrite all the previous one.

Reply:
For me, we do not want to use `/etc/sysctl.conf` so that if one day we want to clean up a server it's really simpler to just remove our files  (cc @sayf-eddine-scality )

Reply:
I am just suggesting u chose.
That is not the first time we face such an issue, we simply don't have a lot of control over drop_in files. either
- we setup our own file which is recommended but we risk ppl or other tools to take precedence
- or we simply add it directly in the root file and maybe add `# added by metalk8s` to be able to clean it later.